trio/_core/_ki.py

```diff
@@ -109,6 +108,14 @@ def currently_ki_protected():
     return ki_protection_enabled(sys._getframe())
 
 
+# functions decorated @async_generator are given this magic property that's a
+# reference to the object itself
+# see python-trio/async_generator/async_generator/_impl.py
+# don't give deprecationwarning since aclosing() is not in stdlib <3.10
+def async_generator_isasyncgenfunction(obj):
```

Comment:
perhaps naming it `legacy_isasyncgenfunction()` would be more clear

comment should clearly state this is supporting the async_generator backfill package